### ISSUE TITLE
"AdminLinks" extension is no longer able to detect the special pages provided by Semantic MediaWiki

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticMediaWiki#1546]
### Setup and configuration

- PHP = 5.4
- MySQL = 5.5
- MW = 1.26.2
- AdminLinks = [master](https://git.wikimedia.org/tree/mediawiki%2Fextensions%2FAdminLinks.git/31b1924e3d25be3356414231da971a7690ab8741)
- SMW = master (f26bc07)

### Issue

The "AdminLinks" extension is no longer able to detect the special pages provided by Semantic MediaWiki so [this code](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/d6011d08f1cccc857432488b38c56374a7d8762c/SemanticMediaWiki.hooks.php#L32-L82) seems to be broken or being ignored.

After the upgrade of sandbox I will be able to show the issue live.

See also: [T134077](https://phabricator.wikimedia.org/T134077)

**Comment by JeroenDeDauw** [comment]
Ping @yaronkoren

**Comment by yaronkoren** [comment]
Hi - I think it's a problem in SMW (and not Admin Links). Specifically, this hook function is apparently getting called before the special pages are defined.

**Comment by JeroenDeDauw** [comment]
The hook is only invoked by whatever calls the callbacks in `$wgHooks['AdminLinks']`. Why would this now be before the special pages are defined?

AFAIK no relevant code in SMW changed, so something somewhere else must have.

**Comment by mwjames** [comment]
> AFAIK no relevant code in SMW changed, so something somewhere else must have.

Exactly.

My field test shows:

```
wfLoadSkin( 'Vector' );
wfLoadExtension( 'SemanticForms' );

# End of automatically generated settings.
# Add more configuration options below.

require_once( "$IP/extensions/AdminLinks/AdminLinks.php" );

/**
 * SMW
 */
enableSemantics( 'http://example.org/id/', true );
```

![image](https://cloud.githubusercontent.com/assets/1245473/15036417/c234388e-128e-11e6-8460-caad37bdf65b.png)

![image](https://cloud.githubusercontent.com/assets/1245473/15036425/e24c6e52-128e-11e6-811b-50b35406bc51.png)

In light of above findings, I'm compelled to close this as not related to SMW (based on the above evidence).

I am not claiming that here is no issue but the relevant code paths are executed for the extensions installed which means that from a SMW-code perspective (and for the execution of the hook) there is nothing left to do. Whether MW's extension registration interferes and is responsible for the disappearance I'm unable to say.

>  I think it's a problem in SMW (and not Admin Links). Specifically, this hook function is apparently getting called before the special pages are defined.

I think in the interest of all parties, providing evidence is much more helpful then making claims of arbitrary value.

**Comment by kghbln** [comment]
Perhaps MW is the issue. MW 1.25 is cool and MW 1.28 seems to be cool. My wiki is broken with MW 1.26 and I did not even dare install MW 1.27. I will try to find some working combination. As last resort I can still pull AdminLinks from the wikis since it lost its usefulness with this issue. Keeping fingers crossed.

**Comment by kghbln** [comment]
I come to conclude that this issue https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/1424 already kicks in with MW 1.26.x so it is apparently a MediaWiki issue fixed with MW 1.27
